tests/cpp/analysis/bls_analyzer_test.cpp

```diff
@@ -74,6 +88,68 @@ TEST_F(BLSAnalyzerTest, Basic) {
   EXPECT_EQ(pad.bounds[1].high, kBlockSize);
 }
 
+TEST_F(BLSAnalyzerTest, Mul) {
```

Comment:
nit: Since we mostly want to test `DiffRange`'s multiplication and SHL, we can create a new test just for `DiffRange`. But I think these are fine to have :-)

Reply:
Alright. yeah we don't seem to have any tests specifically for DiffRange yet. I'll figure out how to make some.

Reply:
@k-ye tests add!

Reply:
Awesome. FYI one of our interns will test this PR with https://github.com/taichi-dev/taichi_elements next week. Let's see how well it works (or what we are missing...)!